pandas/core/groupby/generic.py

```diff
@@ -1894,12 +1894,10 @@ def _normalize_keyword_aggregation(kwargs):
     """
     # Normalize the aggregation functions as Mapping[column, List[func]],
     # process normally, then fixup the names.
-    # TODO(Py35): When we drop python 3.5, change this to
-    # defaultdict(list)
     # TODO: aggspec type: typing.OrderedDict[str, List[AggScalar]]
     # May be hitting https://github.com/python/mypy/issues/5958
     # saying it doesn't have an attribute __name__
-    aggspec = OrderedDict()
+    aggspec = defaultdict(list)
```

Comment:
The change is more complex than this. The code below can be simplified if this is a defaultdict.

Also, is `OrderedDict` still being used after this, or can it be removed from the imports?

Reply:
Quite a few other uses of OrderedDict -> i'm aiming to eliminate most in this issue https://github.com/pandas-dev/pandas/issues/30469

Reply:
> Also, is `OrderedDict` still being used after this, or can it be removed from the imports?

Yes OrderedDIct is still being used.

I am really new to developing, can you please explain what is the problem with ```OrderedDict``` ?

Reply:
With a regular dict:
```python
groupping_values = {}
for value in [1, 3, 1, 1, 2, 3]:
    if value not in groupping_values:
        groupping_values[value] = [value]
    else:
        groupping_values[value].append(value)
```
The if is needed, because when we create the key in the dictionary, we need to create the list with its value. The other times we can append to it. 

With an example, if we have an empty dictionary `{}` and we want to add the Python programming language, we can do `stuff['languages'] = ['python']`. If now that we already have `languages` with a list, if we want to add javascript we use a different instruction `stuff['languages'].append('javascript')`. This is what the `if` in the example does.

Python `defaultdict` exists to make things simpler:
```python
import collections
groupping_values = collections.defaultdict(list)
for value in [1, 3, 1, 1, 2, 3]:
    groupping_values[value].append(value)
```

A `defaultdict` is "smart", since in the creation we tell its elements will be list. So, when we call the append, it'll create an empty list automatically before. We don't need the `if` anymore, and the code is simpler.

This is what it's expected here.

Reply:
Thank you so much for the detailed explanation.